Sources/Utility/system.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import enum POSIX.Error
+import func POSIX.getenv
+import Foundation
+
+/**
+ Execute the specified arguments where the first argument is
+ the tool. Uses PATH to find the tool if the first argument
+ path is not absolute.
+ */
+public func system(_ args: String...) throws {
+    try system(args)
+}
+
+/**
+ Execute the specified arguments where the first argument is
+ the tool. Uses PATH to find the tool if the first argument
+ path is not absolute.
+ */
+public func system(_ arguments: [String], environment: [String:String] = [:]) throws {
+    printArgumentsIfVerbose(arguments)
+
+    let task = NSTask()
+
+    var arguments = arguments
+    task.launchPath = which(arguments.removeFirst())
+    task.arguments = arguments
+
+    var environment = environment
+    #if Xcode
+        let keys = ["SWIFT_EXEC", "HOME", "PATH"]
```

Comment:
This lost the change to add: `"TOOLCHAINS", ` here that I made in the past couple of days.